deps/ncrypto/ncrypto.cc

```diff
@@ -1754,4 +1800,166 @@ EVPKeyPointer::ParseKeyResult EVPKeyPointer::TryParsePrivateKey(
   };
 }
 
+Result<BIOPointer, bool> EVPKeyPointer::writePrivateKey(
+    const PrivateKeyEncodingConfig& config) const {
+  if (config.format == PKFormatType::JWK) {
+    return Result<BIOPointer, bool>(false);
+  }
+
+  auto bio = BIOPointer::NewMem();
+  if (!bio) {
+    return Result<BIOPointer, bool>(false);
+  }
+
+  auto passphrase = GetPassphrase(config);
+  MarkPopErrorOnReturn mark_pop_error_on_return;
+  bool err;
+
+  switch (config.type) {
+    case PKEncodingType::PKCS1: {
+      // PKCS1 is only permitted for RSA keys.
+      if (id() != EVP_PKEY_RSA) return Result<BIOPointer, bool>(false);
+
+#if OPENSSL_VERSION_MAJOR >= 3
```

Comment:
Can we make this a function as well?

Reply:
Make what a function? The macro check? Not sure why?

Reply:
That said, as this moves forward, a lot of this is going to be refactored. Currently it follows a lot of the same pattern as the existing code with some cleanups but this will continue to evolve. I'd rather keep it as it for now and tackle more cleanups in another iteration